From 11565f2cd413aad0566435746a38ea21984ed1a8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marc-Andr=C3=A9=20Lureau?= <marcandre.lureau@redhat.com>
Date: Wed, 22 Jan 2014 18:37:19 +0100
Subject: [PATCH] session: don't track open_host_idle source id

In all cases, when the coroutine is resumed, the idle source has been
running and thus will be removed. Doing it a second time results in the
wrong source being removed or an invalid source warning.

Related: rhbz#1097338

(cherry picked from commit 39555c4f69261c2297c99c75885f94c8daae7216)
---
 gtk/spice-session.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/gtk/spice-session.c b/gtk/spice-session.c
index bcbba27..9e6c154 100644
--- a/gtk/spice-session.c
+++ b/gtk/spice-session.c
@@ -1825,10 +1825,9 @@ GSocketConnection* spice_session_channel_open_host(SpiceSession *session, SpiceC
         g_timeout_add_seconds(SOCKET_TIMEOUT, connect_timeout, &open_host);
 #endif
 
-    guint id = g_idle_add(open_host_idle_cb, &open_host);
+    g_idle_add(open_host_idle_cb, &open_host);
     /* switch to main loop and wait for connection */
     coroutine_yield(NULL);
-    g_source_remove(id);
 
 #if !GLIB_CHECK_VERSION(2,26,0)
     if (open_host.timeout_id == 0)
